devtools/src/local-storage.provider.ts

```diff
@@ -0,0 +1,16 @@
+/**
+ * @license
+ * Copyright Google LLC All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.dev/license
+ */
+
+import {InjectionToken, Provider} from '@angular/core';
+
+export const LOCAL_STORAGE = new InjectionToken<typeof localStorage>('LOCAL_STORAGE');
+
+export const provideLocalStorage = (): Provider => ({
+  provide: LOCAL_STORAGE,
+  useValue: localStorage,
+});
```

Comment:
we could provide a default factory to the token :
```suggestion
export const LOCAL_STORAGE = new InjectionToken<typeof localStorage>('LOCAL_STORAGE', { factory: () => localStorage });
```

This way the token behaves like a providedIn root service.